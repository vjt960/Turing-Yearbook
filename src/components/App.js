import React, { Component } from 'react';
import Cohort from './Cohort';
import people from '../data/yearbook-data.js';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      staff: people.staff, 
      students: people.students
    }
  }

  render() {
    return (
      <div className="App">
      <header className="App-header">
      <h1>Turing Yearbook</h1>
      </header>
      <main className="showcase">
        <Cohort title='Staff' people={this.state.staff} />
        <Cohort title='Students' people={this.state.students} />
      </main>
      </div>
    );
  }
}

export default App;
