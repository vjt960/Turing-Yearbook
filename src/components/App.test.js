import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import App from './App';
import people from '../data/yearbook-data.js';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should hold staff and students in state', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.state('staff')).toEqual(people.staff);
    expect(wrapper.state('students')).toEqual(people.students);
  });

  it('should save a snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  })
})
