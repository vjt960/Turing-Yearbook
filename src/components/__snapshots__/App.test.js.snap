// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should save a snapshot 1`] = `ShallowWrapper {}`;
